src/test/coll.rs

```diff
@@ -545,30 +502,26 @@ async fn find_allow_disk_use_not_specified() {
     let coll = event_client
         .database(function_name!())
         .collection(function_name!());
+    coll.find(None, options).await.unwrap();
 
-    let find_opts = FindOptions::builder().build();
-    coll.find(None, find_opts).await.unwrap();
-
-    assert!(event_client
-        .command_events
-        .read()
-        .unwrap()
-        .iter()
-        .all(|event| match event {
-            CommandEvent::CommandStartedEvent(CommandStartedEvent {
-                command_name,
-                command,
-                ..
-            }) => {
-                if command_name == "find" {
-                    match command.get_bool("allowDiskUse") {
-                        Ok(_) => false,
-                        _ => true,
-                    }
-                } else {
-                    true
-                }
+    let events = event_client.command_events.read().unwrap();
+    let mut iter = events.iter().filter(|event| match event {
+        CommandEvent::CommandStartedEvent(CommandStartedEvent { command_name, .. }) => {
+            command_name == "find"
+        }
+        _ => false,
+    });
+
+    let event = iter.next().unwrap();
+    let allow_disk_use = match event {
+        CommandEvent::CommandStartedEvent(CommandStartedEvent { command, .. }) => {
+            match command.get_bool("allowDiskUse") {
+                Ok(b) => Some(b),
+                Err(_) => None,
             }
-            _ => true,
-        }));
+        }
+        _ => None,
+    };
+    assert_eq!(allow_disk_use, expected_value);
+    assert_eq!(iter.count(), 0);
```

Comment:
One thing that might not be obvious here is that if for some reason there are a large number of other events in the iterator, `count` will have to iterate over all of them in order to complete. A cheaper way of doing this would be to do `assert!(iter.next().is_none())`. I think it's fine to leave it as-is though, since this will give us an error message that tells us exactly how many extra find events there are rather than only the fact that there are some extras.

Reply:
Ah makes sense, I originally had `assert_eq!(iter.next(), None)` but that wasn't compiling so I switched it over to use count, didn't realize there was an `is_none()` method. I'll leave it, I agree it would be useful to see how many extra events are created.

Reply:
The reason that didn't work is that we haven't implemented the equality traits on the `CommandStartedEvent` type. In Rust, the `==` method is only defined for types which implement [`PartialEq`](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html), which can also be derived if all the contained types implement it. There's also [`Eq`](https://doc.rust-lang.org/std/cmp/trait.Eq.html), which can only be implemented on types that implement `PartialEq`; it doesn't add any new functionality, but it informs the compiler that the type implements full equality, which is can use to optimize things.